collector/generic_collector.go

```diff
@@ -83,12 +87,18 @@ func NewCollector(collectorName string, configFile []byte) (*GenericCollector, e
 		minPollFrequency = minSupportedFrequency
 	}
 
+	if len(configInJSON.MetricsConfig) > metricCountLimit {
+		return nil, fmt.Errorf("Too many metrics defined: %d limit: %d", len(configInJSON.MetricsConfig), metricCountLimit)
```

Comment:
My impression was that it would still calculate regular metrics even if something was broken with the application/custom metrics. But apparently the whole monitoring collapse if the application metrics are bad and collector is not created. I can either remove this:

https://github.com/google/cadvisor/blob/master/manager/manager.go#L787

or remove checks during collector creation and leave only these in housekeeps.